src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,30 @@
+import { useContext, useState } from "react";
+import { ActiveUserContext } from "../contexts/ActiveUserContext";
+import { delCommentById } from "../api";
+
+const DeleteComment = ({author, commentId, setIsDeleted}) => {
+    const {activeUser} = useContext(ActiveUserContext);
+    const [isError, setIsError] = useState(null);
+    const [hasClicked, setHasClicked] = useState(false);
+
+    const delhandler = () => {
+        if (!hasClicked) {
+            setHasClicked(true);
+            delCommentById(commentId)
+            .then(() => {
+                setIsDeleted(true);
+                setIsError(null);
+            })
+            .catch(err => {
+                setIsError(err);
```

Comment:
Nice! You just want to untoggle hasClicked again in the .catch block to allow them to attempt the deletion again if the request has failed.